src/installer/url.go

```diff
@@ -0,0 +1,93 @@
+package installer
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+func (installer *OneAgentInstaller) installAgentFromUrl(targetDir string) error {
+	fs := installer.fs
+	tmpFile, err := afero.TempFile(fs, "", "download")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary file for download: %w", err)
+	}
+	defer func() {
+		_ = tmpFile.Close()
+		if err := fs.Remove(tmpFile.Name()); err != nil {
+			log.Error(err, "failed to delete downloaded file", "path", tmpFile.Name())
+		}
+	}()
+
+	if installer.props.Url != "" {
+		if err := installer.downloadOneAgentViaInstallerUrl(tmpFile); err != nil {
+			return err
+		}
+	} else if installer.props.Version == VersionLatest {
+		if err := installer.downloadLatestOneAgent(tmpFile); err != nil {
+			return err
+		}
+	} else {
+		if err := installer.downloadOneAgentWithVersion(tmpFile); err != nil {
+			return err
+		}
+	}
+
+	var fileSize int64
+	if stat, err := tmpFile.Stat(); err == nil {
+		fileSize = stat.Size()
+	}
+
+	log.Info("saved OneAgent package", "dest", tmpFile.Name(), "size", fileSize)
+	log.Info("unzipping OneAgent package")
+	if err := installer.extractZip(tmpFile, targetDir); err != nil {
+		return fmt.Errorf("failed to unzip file: %w", err)
+	}
+	log.Info("unzipped OneAgent package")
+	return nil
```

Comment:
could be refactored into its own function. Also use errors.New() for new errors. If errors.New() does not have formatting options some function in the `pkg/errors` package will, please double check other error returns as well